### PR TITLE
Blog App: validations, model specs, and n+1 problems

### PR DESCRIPTION
**Pull request by HENRYKC24**
## Done in this pull request are:
Added the following validations:
- For the User model:
1. Name must not be blank.
2. PostsCounter must be an integer greater than or equal to zero.
- For the Post model:
1. Title must not be blank.
2. Title must not exceed 250 characters.
3. CommentsCounter must be an integer greater than or equal to zero.
4. LikesCounter must be an integer greater than or equal to zero.
- Added unit specs for all of your models' methods and validations.
- Added flash messages in the create actions in all your controllers.
- Made sure that the N+1 problem is solved when fetching all posts and their comments for a user.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find(params[:user_id])
+    @user = User.includes(:posts).find(params[:user_id])
```
Comment (SajjadAhmad14) [inline app/controllers/posts_controller.rb:3]: Please solve the n+1 problem when fetching all posts and their comments for a user in the `posts#index`. For example:
@posts = @user.posts.includes(:comments)

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -1,5 +1,32 @@
 require 'rails_helper'
 
 RSpec.describe User, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  describe 'Validaiton tests for user' do
+    subject { User.new(name: 'Henry Kc', photo: 'profile.png', bio: 'Henry the great', posts_counter: 0) }
+    before { subject.save }
+
```
Comment (SajjadAhmad14) [inline spec/models/user_spec.rb:7]: Please add unit test for `#last_recent_posts` of User model. It's a requirement to test all custom methods of a model.
rq

File: spec/models/post_spec.rb [hunk 3]
```diff
@@ -1,5 +1,39 @@
...
 RSpec.describe Post, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  describe 'Validaiton for post' do
+    subject do
+      henry = User.new(name: 'Henry Kc', photo: 'profile.png',
+                       bio: 'I am coming out to be the best computer programmer', posts_counter: 0)
+      Post.new(title: 'My best friend', text: 'He will be unvailed at the end of my program at Microverse',
+               comments_counter: 3, likes_counter: 3, author_id: henry.id)
+    end
+
```
Comment (SajjadAhmad14) [inline spec/models/post_spec.rb:11]: Please add a unit test for the custom method which shows the last five comments of the Post model.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Henry, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (HENRYKC24) [reply app/controllers/posts_controller.rb:3]: Thank you for pointing that out.

**Review by rukundoeric — APPROVED** [review]
Hi @HENRYKC24,

_Excellent job on this project. Keep it up 🎆_

_Congratulations! 🎉  on the splendid job 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_


Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

------
>As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**